### PR TITLE
Reorganize the color palette

### PR DESCRIPTION
**Pull request by bl-nero**
This change puts together color combinations that are defined by the [new design system](https://www.figma.com/design/Gpjs9vjhzUKF1GDbeG9JGE/Application-Design-System?node-id=6786-30523&t=32pdAkte34JyVzTw-0). Most of these colors were already there, but scattered and disorganized. This change deprecates the old fields and introduces new ones that have text and background colors grouped together and are intended to serve as the source of truth.

This is a preparation step for the upcoming change that implements the new button system.

There are no visible changes expected.

Note: This change is coupled with https://github.com/gravitational/teleport.e/pull/4457, but (hopefully) not enough to break the build or tests.

### CODE REVIEW

File: web/packages/design/src/theme/themeColors.story.tsx [hunk 1]
```diff
@@ -357,17 +355,20 @@ function ColorsBox({ colors, themeType = null, ...styles }) {
...
-    </Flex>
+    <Box width="150px" height="150px" p={3} mr={3} bg={bg} {...styles}>
+      <Text color={color} css={``}>
+        {/* Path, potentially broken along the periods. */}
+        {path.split('.').map((word, i, arr) => (
+          <>
+            {word}
+            {i < arr.length - 1 ? '.' : ''}
+            {/*potential line break*/}
+            <wbr />
```
Comment (ravicious) [inline web/packages/design/src/theme/themeColors.story.tsx:368]: TIL `<wbr />` exists.

File: web/packages/design/src/theme/themeColors.story.tsx [hunk 2]
```diff
@@ -357,17 +355,20 @@ function ColorsBox({ colors, themeType = null, ...styles }) {
...
-        css={`
-          color: ${props => props.theme.colors.text.slightlyMuted};
-        `}
-      >
-        {path}
-      </Box>
-      <Box width="100%" height="50px" p={3} mr={3} bg={color} {...styles} />
-    </Flex>
+    <Box width="150px" height="150px" p={3} mr={3} bg={bg} {...styles}>
+      <Text color={color} css={``}>
```
Comment (ravicious) [inline web/packages/design/src/theme/themeColors.story.tsx:361]:
```suggestion
      <Text color={color}>
```

File: web/packages/design/src/theme/themes/types.ts [hunk 3]
```diff
@@ -87,13 +104,15 @@ export type ThemeColors = {
     textDisabled: string;
     bgDisabled: string;
 
+    /** @deprecated Use `interactive.solid.primary` instead. */
```
Comment (ravicious) [inline web/packages/design/src/theme/themes/types.ts:107]: Oh, that is huge that this now works properly thanks to types in UI components.

File: web/packages/design/src/theme/utils/colorManipulator.js [hunk 4]
```diff
@@ -133,9 +133,9 @@ export function decomposeColor(color) {
 /**
  * Converts a color object with type and values to a string.
  *
- * @param {object} color - Decomposed color
- * @param {string} color.type - One of: 'rgb', 'rgba', 'hsl', 'hsla'
- * @param {array} color.values - [n,n,n] or [n,n,n,n]
+ * @param {object} bgColor - Decomposed color
+ * @param {string} bgColor.type - One of: 'rgb', 'rgba', 'hsl', 'hsla'
+ * @param {array} bgColor.values - [n,n,n] or [n,n,n,n]
```
Comment (ravicious) [inline web/packages/design/src/theme/utils/colorManipulator.js:138]: Is this a find and replace all incident or is it indeed supposed to be called `bgColor`? ;)

**Review by ravicious — APPROVED** [review]
🚀

File: web/packages/design/src/theme/themes/types.ts (hunk 3 above)
Comment (bl-nero) [reply web/packages/design/src/theme/themes/types.ts:107]: Unfortunately, only in style functions, but the usage of colors in attributes is somehow limited to just a handful of cases.

File: web/packages/design/src/theme/utils/colorManipulator.js (hunk 4 above)
Comment (bl-nero) [reply web/packages/design/src/theme/utils/colorManipulator.js:138]: What the...

**Review by gzdunek — APPROVED** [review]

**Pull request merged**